code/game/gamemodes/game_mode.dm

```diff
@@ -415,6 +416,115 @@
+
+///////////////////////////////////////
+//Keeps track of all science members //
+///////////////////////////////////////
+/datum/game_mode/proc/get_all_sci()
+	. = list()
+	for(var/mob/living/carbon/human/player in GLOB.mob_list)
+		if(player.mind && (player.mind.assigned_role in GLOB.science_positions))
+			. |= player.mind
+
+/////////////////////////////////////////////
+//Keeps track of all living medical members//
+/////////////////////////////////////////////
+/datum/game_mode/proc/get_living_med()
+	. = list()
+	for(var/mob/living/carbon/human/player in GLOB.mob_list)
+		if(player.stat != DEAD && player.mind && (player.mind.assigned_role in GLOB.medical_positions))
+			. |= player.mind
+
+///////////////////////////////////////
+//Keeps track of all medical members //
+///////////////////////////////////////
+/datum/game_mode/proc/get_all_med()
+	. = list()
+	for(var/mob/living/carbon/human/player in GLOB.mob_list)
+		if(player.mind && (player.mind.assigned_role in GLOB.medical_positions))
+			. |= player.mind
+
+///////////////////////////////////////////
+//Keeps track of all living cargo members//
+///////////////////////////////////////////
+/datum/game_mode/proc/get_living_cargo()
+	. = list()
+	for(var/mob/living/carbon/human/player in GLOB.mob_list)
+		if(player.stat != DEAD && player.mind && (player.mind.assigned_role in GLOB.supply_positions))
+			. |= player.mind
+
+/////////////////////////////////////
+//Keeps track of all cargo members //
+/////////////////////////////////////
+/datum/game_mode/proc/get_all_cargo()
+	. = list()
+	for(var/mob/living/carbon/human/player in GLOB.mob_list)
+		if(player.mind && (player.mind.assigned_role in GLOB.supply_positions))
+			. |= player.mind
+
+//////////////////////////////////////////////
+//Keeps track of all living civilian members//
+//////////////////////////////////////////////
+/datum/game_mode/proc/get_living_civ()
+	. = list()
+	for(var/mob/living/carbon/human/player in GLOB.mob_list)
+		if(player.stat != DEAD && player.mind && (player.mind.assigned_role in GLOB.civilian_positions))
+			. |= player.mind
+
+////////////////////////////////////////
+//Keeps track of all civilian members //
+////////////////////////////////////////
+/datum/game_mode/proc/get_all_civ()
+	. = list()
+	for(var/mob/living/carbon/human/player in GLOB.mob_list)
+		if(player.mind && (player.mind.assigned_role in GLOB.civilian_positions))
+			. |= player.mind
+
+/////////////////////////////////////////////
+//Keeps track of all living silicon members//
+/////////////////////////////////////////////
+/datum/game_mode/proc/get_living_silicon()
+	. = list()
+	for(var/mob/living/carbon/human/player in GLOB.mob_list)
+		if(player.stat != DEAD && player.mind && (player.mind.assigned_role in GLOB.nonhuman_positions))
+			. |= player.mind
+
+///////////////////////////////////////
+//Keeps track of all silicon members //
+///////////////////////////////////////
+/datum/game_mode/proc/get_all_silicon()

```

Comment:
This doesn't keep track of all silicons. They aren't of the `mob/living/carbon/human` type, so this proc will always return nothing. Also, we already display all the silicons at round end, you might wanna check that code to see if there is anything you could steal.

Reply:
Addressed with the latest commit, it didn't mark this as outdated because the line is the same.